test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -183,34 +187,29 @@ database_operation::read_operation(thread_context *tc)
 {
     logger::log_msg(
       LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
-    WT_CURSOR *cursor = nullptr;
-    WT_DECL_RET;
+
     std::map<uint64_t, scoped_cursor> cursors;
     while (tc->running()) {
         /* Get a collection and find a cached cursor. */
         collection &coll = tc->db.get_random_collection();
-        const auto &it = cursors.find(coll.id);
-        if (it == cursors.end()) {
-            auto sc = tc->session.open_scoped_cursor(coll.name.c_str());
-            cursor = sc.get();
-            cursors.emplace(coll.id, std::move(sc));
-        } else
-            cursor = it->second.get();
+
+        if (cursors.find(coll.id) == cursors.end())
+            cursors.emplace(coll.id, std::move(tc->session.open_scoped_cursor(coll.name.c_str())));
+
+        /* Do a second lookup now that we know it exists. */
+        auto &it = cursors[coll.id];
 
         tc->transaction.begin();
         while (tc->transaction.active() && tc->running()) {
-            ret = cursor->next(cursor);
-            /* If we get not found here we walked off the end. */
-            if (ret == WT_NOTFOUND) {
-                testutil_check(cursor->reset(cursor));
-            } else if (ret != 0)
-                testutil_die(ret, "cursor->next() failed");
+            if (tc->next(it) == WT_ROLLBACK)
+                /* We got an error, our transaction has been rolled back. */
+                break;
```

Comment:
Can we `continue` instead ?

Reply:
We wouldn't have called begin when we would need to if we `continue` here.

Reply:
We have `try_begin` for that don't we?